src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,963 @@
+                Next
+                DirectoryList = FileSystem.GetDirectories(TestBase.TestDirectory, SearchOption.SearchTopLevelOnly, "*000", "*001")
+                Assert.True(DirectoryList.Count = 2, $"Search results Expected 2 Actual {DirectoryList.Count} DirectoryList = {DirectoryListToString(DirectoryList)} Created List = {DirectoryListToString(New ReadOnlyCollection(Of String)(CreatedDirectories))}")
+                For i As Integer = 0 To 1
+                    Dim DirectoryName As String = IO.Path.Combine(TestBase.TestDirectory, $"NewSubDirectory00{i}")
+                    Assert.True(DirectoryList.Contains(DirectoryName), $"{DirectoryName} Is missing from Wildcard Search")
+                Next
+                IO.Directory.CreateDirectory(IO.Path.Combine(TestBase.TestDirectory, $"NewSubDirectory000", $"NewSubSubDirectory000"))
+                DirectoryList = FileSystem.GetDirectories(TestBase.TestDirectory, SearchOption.SearchTopLevelOnly, "*000")
+                Assert.True(DirectoryList.Count = 1, $"Search results Expected 1 Actual {DirectoryList.Count} {DirectoryListToString(DirectoryList)}")
+                DirectoryList = FileSystem.GetDirectories(TestBase.TestDirectory, SearchOption.SearchAllSubDirectories, "*000")
+                Assert.True(DirectoryList.Count = 2, $"Search results Expected 2 Actual {DirectoryList.Count} {DirectoryListToString(DirectoryList)}")
+            End Using
+        End Sub
+
+        <Fact>
+        Public Shared Sub GetDirectoryInfo_Directory()
+            Using TestBase As New FileIOTests
+                For i As Integer = 0 To 5
+                    IO.Directory.CreateDirectory(IO.Path.Combine(TestBase.TestDirectory, $"NewSubDirectory{i}"))
+                Next
+                IO.Directory.CreateDirectory(IO.Path.Combine(TestBase.TestDirectory, $"NewSubDirectory0", $"NewSubSubDirectory"))
+                Dim info As IO.DirectoryInfo = FileSystem.GetDirectoryInfo(TestBase.TestDirectory)
+                Dim infoFromIO As IO.DirectoryInfo = New IO.DirectoryInfo(TestBase.TestDirectory)
+                Assert.True(info.CreationTime = infoFromIO.CreationTime, $"Creation Time info ({info.CreationTime}) <> IO.DriveInfo.CreateTime {infoFromIO.CreationTime})")
+                Assert.True(info.Extension = infoFromIO.Extension, $"Extension {info.Extension} <> IO.DriveInfo.Extension {infoFromIO.Extension}")
+                Assert.True(info.FullName = TestBase.TestDirectory, $"Fullname {info.FullName} <> TestBase.TestDirectory {TestBase.TestDirectory}")
+                Assert.True(info.LastAccessTime = infoFromIO.LastAccessTime)
+                Assert.True(info.Name = infoFromIO.Name, $"Name {info.Name} Doesn't match {infoFromIO.Name}")
+                Assert.True(RemoveEndingSeparator(info.Parent.ToString) = RemoveEndingSeparator(infoFromIO.Parent.ToString), "info.Parent<>infoFromIO.Parent")
```

Comment:
Is `RemoveEndingSeparator` necessary? `FileSystem.GetDirectoryInfo()` should return exactly the same value as `New DirectoryInfo()` so we should be able to compare without stripping of any separators. Then the `RemoveEndingSeperator` helper can be removed.

Reply:
According to the comments it is needed but I will test without it before deleting. It may only be needed for cases I don't test. In any case I sound use my new TrimSeparators.

Reply:
The comments in reference source but at least in all my tests it is not needed so I deleted. It is needed to handle some issues like C:\Dir1////\\\. The difference between calling IO.SomeFunction and the VB equivalent is dealing with final separator. The VB libraries delete the trailing slashes and IO.SomeFunction leaves them. You see code like below throughout reference Source.
``` VB
IO.Path.GetDirectoryName(path.TrimEnd(IO.Path.DirectorySeparatorChar, IO.Path.AltDirectorySeparatorChar))

 Return GetLongPath(RemoveEndingSeparator(IO.Path.GetFullPath(Path)))
```